coreneuron/nrniv/cn_parameters.h

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) 2016, Blue Brain Project
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef CN_PARAMETERS_H
+#define CN_PARAMETERS_H
+
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+#include "coreneuron/utils/CLI11/CLI.hpp"
+
+/**
+ * \class cn_parameters
+ * \brief Parses and contains Command Line parameters for Core Neuron
+ *
+ * This structure contains all the parameters that CoreNeuron fetches
+ * from the Command Line. It uses the CLI11 libraries to parse these parameters
+ * and saves them in an internal public structure. Each parameter can be
+ * accessed or written freely. By default the constructor instantiates a
+ * CLI11 object and initializes it for CoreNeuron use.
+ * This object is freely accessible from any point of the program.
+ * An ostream method is also provided to print out all the parameters that
+ * CLI11 parse.
+ * Please keep in mind that, due to the nature of the subcommands in CLI11,
+ * the command line parameters for subcategories NEED to be come before the relative
+ * parameter. e.g. --mpi --gpu gpu --nwarp
+ * Also single dash long options are not supported anymore (-mpi -> --mpi).
+ */
+
+typedef std::string string_t;
+
+struct cn_parameters {
+
+    int spikebuf=100000;           /// internal buffer used on every rank for spikes
```

Comment:
I think it would be nice all the variables to have descriptions in the same place

Reply:
Not all the variables had a description and for most of them I had no idea what they did. I can certainly ask and complete the descriptions.

Reply:
I see.. You can add whatever you can find :+1: